### PR TITLE
Adds the Pyromania character trait

### PR DESCRIPTION
**Pull request by ghost**
Inspired by personal ideas and idle chatter in discord. Now you can roleplay the Trashcan Man from The Stand.
* Earns 2 points. Available only in character creation (not because it's important to keep it there, just because I don't think it fits anywhere else.)
* Every two hours spent awake that you haven't lit a fire in gives a stacking, permanent mood detriment, up to a maximum of -30.
* Lighting a fire will cleanse this mood detriment and add a mood increase of +5 to a maximum of +10 that lasts for 24 hours and begins decaying after 16 hours.
  * There's a 25% chance that lighting a fire won't be enough to satisfy your urges, and you'll need to light another one.
* Standing near a fire that won't hurt you gives a mood bonus of 5 for 30 minutes. Standing near a fire that *does* hurt you gives a mood bonus of 10 for an hour, due to the rush of being so close to such intense heat!
* Using a flamethrower weapon instantly fills up your lighting-a-fire mood bonus to +25 as long as you're currently suffering from fire withdrawal.

I tried to model this after realistic pyromania, which, to my knowledge, is a compulsion to light fires, not a neurotic, irresistible desire to Burn All The Things Ever (though that might be a fantasy of theirs.) I also deliberately avoided lighting random fires, due to how unfun that is as a design.

### CODE REVIEW

File: data/json/mutations.json [hunk 1]
```diff
@@ -5050,6 +5050,13 @@
         "prereqs" : ["ACIDPROOF"],
         "threshreq" : ["THRESH_INSECT", "THRESH_CHIMERA", "THRESH_MEDICAL", "THRESH_SLIME"],
         "category" : ["MUTCAT_INSECT", "MUTCAT_CHIMERA", "MUTCAT_MEDICAL", "MUTCAT_SLIME"]
+    },{
+        "type" : "mutation",
+        "id" : "PYROMANIA",
+        "name" : "Pyromania",
+        "starting_trait" : true,
+        "points" : 1,
+        "description" : "Fire has always enthralled and fascinated you, and you gain a mood boost from lighting and being near them."
```
Comment (DracoGriffin) [inline data/json/mutations.json]: Suggest `you gain a mood boost from lighting and being near them.` -> `you gain a boost to mood from igniting and proximity to them.`

File: src/iuse_actor.cpp [hunk 2]
```diff
@@ -1024,10 +1025,16 @@ bool firestarter_actor::prep_firestarter_use( const player &p, tripoint &pos )
...
-void firestarter_actor::resolve_firestarter_use( const player &p, const tripoint &pos )
+void firestarter_actor::resolve_firestarter_use( player &p, const tripoint &pos )
 {
     if( g->m.add_field( pos, fd_fire, 1, 10_minutes ) ) {
-        p.add_msg_if_player( _( "You successfully light a fire." ) );
+        if( !p.has_trait( trait_PYROMANIA ) ) {
+            p.add_msg_if_player( _( "You successfully light a fire." ) );
+        }
+        else {
+            p.add_msg_if_player( m_good, _( "You happily light a fire." ) );
```
Comment (DracoGriffin) [inline src/iuse_actor.cpp]: Suggest `You happily light a fire.` -> `Setting that fire felt good... really good.`

**Comment by DracoGriffin** [comment]
I don't agree with the higher mood boost in fire compared to outside of fire, as I can't seem to find anything to support pyromaniacs self-immolating (even small parts of body).

Also, I think it'd be interesting to have withdrawal symptoms if gone without lighting a fire within a time frame, maybe 24/48 hours or something? `You really wanna burn something.` -> `You NEED to set fires, man, and watch stuff burn.`

Lastly, if possible, a third major mood boost from using flamethrowers and equivalent.

File: src/trapfunc.cpp [hunk 3]
```diff
@@ -839,6 +841,10 @@ void trapfunc::lava( Creature *c, const tripoint &p )
             n->deal_damage( nullptr, bp_foot_r, damage_instance( DT_HEAT, 20 ) );
             n->deal_damage( nullptr, bp_leg_l, damage_instance( DT_HEAT, 20 ) );
             n->deal_damage( nullptr, bp_leg_r, damage_instance( DT_HEAT, 20 ) );
+            if( n->has_trait( trait_PYROMANIA ) ) {
+                n->rem_morale( MORALE_PYROMANIA_NEARFIRE );
+                n->add_morale( MORALE_PYROMANIA_MOREDOTS, 100, 100, 1_hours, 30_minutes ); // we're probably gonna die here, but might as well die happy, right?
```
Comment (ZhilkinSerg) [inline src/trapfunc.cpp]: Please move comment to separate line as it makes line too long.

File: src/iuse_actor.cpp [hunk 4]
```diff
@@ -1024,10 +1025,16 @@ bool firestarter_actor::prep_firestarter_use( const player &p, tripoint &pos )
...
 
-void firestarter_actor::resolve_firestarter_use( const player &p, const tripoint &pos )
+void firestarter_actor::resolve_firestarter_use( player &p, const tripoint &pos )
 {
     if( g->m.add_field( pos, fd_fire, 1, 10_minutes ) ) {
-        p.add_msg_if_player( _( "You successfully light a fire." ) );
+        if( !p.has_trait( trait_PYROMANIA ) ) {
+            p.add_msg_if_player( _( "You successfully light a fire." ) );
+        }
+        else {
```
Comment (ZhilkinSerg) [inline src/iuse_actor.cpp]: `} else {` should be on same line

**Comment by nexusmrsep** [comment]
First of all, I think this is a great idea @Xhuis.

I realy like @DracoGriffin suggestions on this PR. Also the sheer fact of igniting a fire, lighting an oil lamp, using a lighter, matchbook, etc. should in this case give a morale bonus and satisfy some of those 'addiction-like' needs.

Also - as this is an mental disorder - you might even compulsively (automatically) try to randomly light your environment if you have a lighter/matchbook etc. on you (near you) and you are in withdrawal. As a result you would need to light a fire now and then to keep your pyromania at bay, else you will eventually ignite your own bed.

**Comment by kevingranade** [comment]
Check #1113 for previous discussion on a similar feature.
Only positive morale effects is a problem, especially given the magnitude of the bonuses.
A pyromaniac that never interacts with fires should have heavy morale penalties.

**Comment by Xhuis** [comment]
Would you say it's best, then, to keep it as a positive trait with primarily good factors, or as a negative/mixed trait with benefits if managed correctly and downsides if not?

I'm also very much against starting random fires. My experience in other games has taught me that losing a great deal or causing irreparable damage based on random chance from small mismanagement isn't fun design.

**Comment by kevingranade** [comment]
I prefer it being a negative trait that can be gamed to be positive.

I dont like the random fires thing either, that can be almost immediately
fatal.

>

**Comment by DracoGriffin** [comment]
>Also - as this is a mental disorder - you might even compulsively (automatically) try to randomly light your environment if you have a lighter/matchbook etc. on you (near you) and you are in withdrawal. As a result you would need to light a fire now and then to keep your pyromania at bay, else you will eventually ignite your own bed.

During some brief research on Discord, this behavior is extremely rare and basically occurs with younger sufferers of the disorder because they don't have complete awareness of the ramifications of starting fires. Generally, as adults (which [have an extremely](https://books.google.com/books?id=dCwpAiGycekC&lpg=PA260&ots=LVH6LLxzQD&dq=pyromania%20prevalence%20rate&pg=PA260#v=onepage&q=pyromania%20prevalence%20rate&f=false) low [prevalence rate](http://www.oxfordhandbooks.com/view/10.1093/oxfordhb/9780195389715.001.0001/oxfordhb-9780195389715-e-011), firesetting is much lower than as children/adolescents. Secondly, it's also estimated that most pyromaniacs are not arsonists but do engage in firesetting. [This link](http://jaapl.org/content/40/3/355.long) I believe helps clear out a lot of the improperly used words and actions. If no one else can access it, I'll try to pick out some relevant quotes rather than extending this comment into a huge rant.

In terms of the minor firesetting (lamps and such), I think it'd be interesting if it randomized between good and bad morale changes (`<good>Lighting that lamp released something in you</good>` or `<bad>This fire isn't big enough... you HAVE to burn something larger!</bad>`), with perhaps some weighted based on recent firesetting (so if you lit a corpse or tree or something on fire or sprayed a flamethrower, you're "satisfied" for awhile and get mainly good messages when lighting small things, compared to going without lighting a fire for 48 hours and you start jonesin' to burn something).

**Comment by nexusmrsep** [comment]
Im not forcing any idea here, and I agree that it would be a hell of a risk, I just find the prospect of a gameplay with a rogue mentaly unstable character that you'd have to watch carefully and satisfy his/her dark (fiery?) needs in order to save his life, as totaly amusing.

**Comment by Xhuis** [comment]
Sorry for being slow on this, I've been slightly preoccupied. I'll get some work done on it shortly.

**Comment by Xhuis** [comment]
The trait has been updated to a -2 point negative trait. Details are in the updated OP, but the summary is:
* Every 2 hours while not asleep, the player gains a -1 mood detriment, up to a max of -30
* Lighting a fire increases mood by 5 and purges the detriment, up to a max of +10 mood, for 24 hours
* Standing near a fire increases mood by 5 for 30 minutes; if it's close enough to possibly harm you, the excitement of the heat gives a mood bonus of 10 for an hour instead
* Using a flamethrower weapon instantly sets your "light a fire" mood bonus to 25 as long as you didn't already have one
* The mood detriment will only happen if both the "near a fire" and "light a fire" bonuses aren't active

This makes the trait into something similar to Asthmatic, but less severe, more predictable, and less convenient to treat. Being mindful and managing the downsides will cause you a slight bonus the majority of the time, but losing track of time means that you'll be craving a fire and steadily growing more desperate to light one.

Numbers are up in the air still.

**Comment by kevingranade** [comment]
> Sorry for being slow on this, I've been slightly preoccupied.

No need to apologize, we get things done when we can.

**Comment by DracoGriffin** [comment]
Looks like the syntax isn't correct and causing translation error as well.

```cplusplus
21:30:55 src/player.cpp:5664:39: error: no matching function for call to '_'
21:30:55             add_msg_if_player( m_bad, _( smokin_hot_fiyah ) );
21:30:55                                       ^
21:30:55 src/translations.h:44:20: note: candidate function not viable: no known conversion from 'std::string' (aka 'basic_string<char>') to 'const char *' for 1st argument
21:30:55 inline const char *_( const char *msg )
21:30:55                    ^
21:30:55 1 error generated.
21:30:55 Makefile:775: recipe for target 'obj/player.o' failed

**Comment by Night-Pryanik** [comment]
> Standing near a fire that won't hurt you gives a mood bonus of 5 for 30 minutes.

It seems that this doesn't work like it should.

`} else if( pyromania && fire_warmth >= 1 ) { // Only give us fire bonus if there's actually fire`

If I'm not mistaken, `fire_warmth` becomes higher than zero only when standing _on_ tile with fire or hot hot air.

**Pull request merged**